src/TesApi.Web/PagedInterfaceExtensions.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Batch;
+using Microsoft.Azure.Management.ResourceManager.Fluent.Core;
+using Polly.Retry;
+
+// TODO: move this to Common.csproj?
+namespace TesApi.Web
+{
+    /// <summary>
+    /// Extension methods and implementations for enumerating paged enumeration/collection types from Azure
+    /// </summary>
+    public static class PagedInterfaceExtensions
```

Comment:
Is this truly necessary?  Again concerned about the net risk this adds to project defining our own complex types and extensions

Reply:
I couldn't find anything analogous elsewhere, and this is the most straightforward of these three complex type modules.